src/app/screens/nftCollection/index.tsx

```diff
@@ -110,9 +115,59 @@ const StyledGridContainer = styled(GridContainer)`
   width: 100%;
 `;
 
+function IsVisibleOrPlaceholder({ children }: PropsWithChildren) {
+  const nodeRef = useRef<HTMLDivElement>(null);
+  const isVisible = useIsVisible(nodeRef, { once: false });
+
+  return (
+    <div ref={nodeRef}>
+      {isVisible ? (
+        children
+      ) : (
+        <CollectibleCollectionGridItem item={{}} itemId="">
+          <NftImage />
+        </CollectibleCollectionGridItem>
+      )}
+    </div>
+  );
+}
+
+function CollectionGridItemWithData({
+  nft,
+  collectionData,
+  isGalleryOpen,
+}: {
+  nft: NonFungibleToken;
+  collectionData: StacksCollectionData;
+  isGalleryOpen: boolean;
+}) {
+  const { data: nftData } = useNftDetail(nft.identifier);
```

Comment:
if i'm not wrong the API call will fail when is bns contract. should we prevent the call?